vllm/core/scheduler.py

```diff
@@ -1347,10 +1348,25 @@ def _get_num_new_tokens(self, seq_group: SequenceGroup,
         for seq in seqs:
             num_new_tokens += seq.get_num_new_tokens()
         assert num_new_tokens > 0
-        # Chunk if a running request cannot fit in.
-        # If number of seq > 1, it means it is doing beam search in a
-        # decode phase. Do not chunk in that case.
+        # Chunk if a running request cannot fit in the given budget.
+        # If number of seq > 1, it means it is doing beam search
+        # in a decode phase. Do not chunk.
         if enable_chunking and len(seqs) == 1:
-            num_new_tokens = min(num_new_tokens,
-                                 budget.remaining_token_budget())
+            remaining_token_budget = budget.remaining_token_budget()
```

Comment:
btw, you mentioned scheduler currently includes token budgets although it was prefix cached. do you plan to fix it?

Reply:
Yeah but it needs more work. Specifically we have to
1. Schedule a sequence.
2. Allocate blocks / mark computed blocks / determine the real new tokens.
3. Update the budget.
4. Schedule the next sequence.

The problem is step 1, 3 and 4 happens in the same function (`_schedule`) but step 2 happens outside it, so this would need some refactoring. It might also be a good idea to work on it after deprecating block manager v1, so that we don't have to deal with two different mark computed logic in scheduler anymore.

Reply:
yeah timeline (after removing block manager v1) sounds good to me. are we planning to do this soon btw?

Reply:
Hopefully...